db.go

```diff
@@ -1430,6 +1430,16 @@ func (db *DB) startMemoryFlush() {
 // stopped. Ideally, no writes are going on during Flatten. Otherwise, it would create competition
 // between flattening the tree and new tables being created at level zero.
 func (db *DB) Flatten(workers int) error {
+	return db.flattenInternal(workers, true)
+}
+
+// CompactLSM works like Flatten except it only performs one pass of compactions instead of
+// performing compactions until the whole database is in a single level like Flatten does.
+func (db *DB) CompactLSM(workers int) error {
```

Comment:
This call could take a very long time if the LSM tree has a lot of data and since compactions are stopped, we might end up accumulating too much data on level 0.

We should add a timeout to this function. 5 or 10 seconds timeout. @ashish-goswami thoughts?